src/recon.rs

```diff
@@ -2521,18 +2521,15 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                 let dst: *mut BD::Pixel = (f.cur.data.data[0] as *mut BD::Pixel).offset(
                     (4 * (t.by as isize * BD::pxstride(f.cur.stride[0]) + t.bx as isize)) as isize,
                 );
-                let pal_idx: *const u8;
-                if t.frame_thread.pass != 0 {
+                let pal_idx = if t.frame_thread.pass != 0 {
                     let p = t.frame_thread.pass & 1;
-                    if ((*ts).frame_thread[p as usize].pal_idx).is_null() {
-                        unreachable!();
-                    }
-                    pal_idx = (*ts).frame_thread[p as usize].pal_idx;
-                    (*ts).frame_thread[p as usize].pal_idx =
-                        ((*ts).frame_thread[p as usize].pal_idx).offset((bw4 * bh4 * 16) as isize);
+                    let frame_thread = &mut (*ts).frame_thread[p as usize];
+                    let pal_idx = &f.frame_thread.pal_idx[frame_thread.pal_idx..];
+                    frame_thread.pal_idx += (bw4 * bh4 * 16) as usize;
```

Comment:
```suggestion
                    let len = (bw4 * bh4 * 16) as usize;
                    let pal_idx = &f.frame_thread.pal_idx[frame_thread.pal_idx..][..len];
                    frame_thread.pal_idx += len;
```

If this becomes a `DisjointMut` (it looks like it probably would?), we want these borrows to be more fine-grained, so we want to cut off the excess length to only what we use.  So for these `.split_at_mut()` like places, we should add `[..len]`s.